packages/docgen/src/engine.js

```diff
@@ -1,25 +1,27 @@
 /**
  * External dependencies.
  */
-const espree = require( 'espree' );
+const babel = require( '@babel/parser' );
 const { flatten } = require( 'lodash' );
 
 /**
  * Internal dependencies.
  */
 const getIntermediateRepresentation = require( './get-intermediate-representation' );
+const babelConfig = require( '../../../babel.config' );
 
-const getAST = ( source ) =>
-	espree.parse( source, {
-		attachComment: true,
-		loc: true,
-		ecmaVersion: 2018,
-		ecmaFeatures: {
-			jsx: true,
-		},
-		sourceType: 'module',
+const getAST = ( source ) => {
+	const { presets, plugins } = babelConfig( {
+		cache: () => {}, // To bypass api.cache is not a function error.
 	} );
 
+	return babel.parse( source || '', {
+		presets,
```

Comment:
It doesn't look like babel-parser admits a [preset option](https://babeljs.io/docs/en/babel-parser/#options)?

Reply:
Right, it should be removed.

Reply:
I remember it didn't work properly without this. I'll check it out by reading @babel/parser code.